src/twisted/internet/tcp.py

```diff
@@ -1038,42 +1285,62 @@ def doRead(self):
+                        continue
+                    elif e.args[0] in (
+                            EMFILE, ENOBUFS, ENFILE, ENOMEM, ECONNABORTED):
+                        # Linux can give ENFILE if the system is out
+                        # of inodes, or ENOMEM if there is
+                        # insufficient memory to allocate a new
+                        # dentry.  ECONNABORTED is documented as
+                        # possible on all relevant platforms (Linux,
+                        # Windows, macOS, and the BSDs) but occurs
+                        # only on the BSDs.  It occurs when a client
+                        # sends a FIN or RST after the server sends a
+                        # SYN|ACK but before application code calls
+                        # accept(2).  On Linux, calling accept(2) on
+                        # such a listener returns a connection that
+                        # fails as though the it were terminated after
+                        # being fully established.  This appears to be
+                        # an implementation choice (see inet_accept in
+                        # inet/ipv4/af_inet.c).  On macOS, such a
+                        # listener is not considered readable, so
+                        # accept(2) will never be called.  Calling
+                        # accept(2) on such a listener, however, does
+                        # not return at all.
                         log.msg("Could not accept new connection (%s)" % (
                             errorcode[e.args[0]],))
                         break
                     raise
 
-                fdesc._setCloseOnExec(skt.fileno())
-                protocol = self.factory.buildProtocol(self._buildAddr(addr))
-                if protocol is None:
-                    skt.close()
-                    continue
-                s = self.sessionno
-                self.sessionno = s+1
-                transport = self.transport(skt, protocol, addr, self, s, self.reactor)
-                protocol.makeConnection(transport)
+                if self._reservedFD.replace(skt):
+                    log.msg(
+                        "EMFILE recovery:"
+                        " Closing socket %r"
+                        " and reopening reserve file descriptor")
+                else:
+                    fdesc._setCloseOnExec(skt.fileno())
+                    protocol = self.factory.buildProtocol(
+                        self._buildAddr(addr))
+                    if protocol is None:
+                        skt.close()
+                        continue
+                    s = self.sessionno
+                    self.sessionno = s+1
+                    transport = self.transport(
+                        skt, protocol, addr, self, s, self.reactor)
+                    protocol.makeConnection(transport)
             else:
                 self.numberAccepts = self.numberAccepts+20
-        except:
+        except BaseException:
```

Comment:
This snuck into the diff so `pycodestyle` complained about it.  Maybe it should be `Exception`.

Reply:
Preserving the old behavior in _this_ PR seems like the wiser choice to me.  So stick with `BaseException` (I think that's equivalent to bare-except?).  Of course, if the goal is not to change the behavior, not changing the code might be the _best_ choice... and if pycodestyle is making a mistake here (complaining about this pre-existing code), it's making a mistake and we should ignore it (after filing a bug report for the mistake).